src/unix/notbsd/linux/mod.rs

```diff
@@ -165,6 +165,20 @@ s! {
         _pad: [::uint8_t; 48],
     }
 
+
+    pub struct sigevent {
+        pub sigev_value: sigval_t,
+        pub sigev_signo: ::c_int,
+        pub sigev_notify: ::c_int,
+        pub sigev_notify_function: ::dox::Option<extern fn(sigev_value: sigval_t)>,
+        pub sigev_notify_attributes: *mut pthread_attr_t,
+        _pad: [u64; 4usize],
+    }
+
+    pub struct sigval_t {
+        _data: [u64; 1usize],
```

Comment:
https://github.com/torvalds/linux/blob/v4.3/include/uapi/asm-generic/siginfo.h#L7-L10

(I have no idea how to handle unions, so can't give any usefl advice)